File: CreationalPatterns/Singleton/C#/Singleton/PizzaOven.cs
public class PizzaOven {
    private static PizzaOven? oven;

    private PizzaOven(){}

    public static PizzaOven GetOven(){
        if(oven == null){
            oven = new PizzaOven();
        }
        return oven;
    }

    public void Bake(){
        Console.WriteLine("Baking in oven");
    }

    public void PreHeat(){
        Console.WriteLine("Preheating oven");
    }
}

File: CreationalPatterns/Singleton/Java/PizzaOven.java
public class PizzaOven {
    private static PizzaOven oven;

    private PizzaOven(){}

    public static PizzaOven getOven(){
        if(oven == null){
            oven = new PizzaOven();
        }
        return oven;
    }

    public void bake(){
        System.out.println("Baking in oven");
    }

    public void preHeat(){
        System.out.println("Preheating oven");
    }
}
